File: src/plugin/npc/MakeoverMageNPCPlugin.java
package plugin.npc;

import com.rs.game.content.PlayerLook;
import com.rs.game.entity.actor.npc.NPC;
import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.type.NPCPlugin;
import com.rs.utility.game.ClickOption;

import static com.rs.utility.game.ClickOption.FIRST;
import static com.rs.utility.game.ClickOption.SECOND;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class MakeoverMageNPCPlugin extends NPCPlugin {
	
	@Override
	public void handle(Player player, NPC npc, ClickOption option) {
		if (option == FIRST) {
			player.getDialogueManager().startDialogue("MakeOverMage", npc.getId(), 0);
		} else if (option == SECOND) {
			PlayerLook.openMageMakeOver(player);
		}
	}
	
	@Override
	public void register() {
		register(2676, FIRST, SECOND);
		register(599, FIRST, SECOND);
	}
}

File: src/main/java/plugin/npc/MakeoverMageNPCPlugin.kt
package plugin.npc

import game.content.entity.actor.player.design.PlayerDesign
import game.content.plugin.type.NPCPlugin
import game.entity.actor.npc.NPC
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/31/2017
 */
class MakeoverMageNPCPlugin : NPCPlugin {
    override fun handle(player: Player, npc: NPC, option: String): Boolean {
        when (option) {
            "Talk-to" -> {
                player.dialogueManager.startDialogue("MakeOverMage", npc.id, 0)
                return true
            }
            "Makeover" -> {
                PlayerDesign.open(player)
                return true
            }
        }
        return false
    }

    override fun register() {
        val ids = intArrayOf(2676, 599)
        for (id in ids) {
            registerNPC(id, "Talk-to")
            registerNPC(id, "Makeover")
        }
    }
}